src/log/setup.cr

```diff
@@ -0,0 +1,39 @@
+class Log
+  # Setups logging bindings discarding all previous configurations.
+  def self.setup(*, builder : Log::Builder = Log.builder)
+    builder.clear
+    yield builder.as(Configuration)
+  end
+
+  # Setups logging for all sources using the specified *level*, *backend*.
+  def self.setup(level : Log::Severity = Log::Severity::Info,
+                 backend : Log::Backend = IOBackend.new,
```

Comment:
Is it possible to have the backend instead be `nil` and use whatever the *existing* backend is (if set) and if none is set *then* set the IOBackend? I'm imagining something like this:

```crystal
if prod?
  Log.setup(:info, some_fancy_backend)
else
  Log.setup(:info, IOBackend.new)
end

Log.setup("db.*", :debug) # I want to use whatever is specified in the first `Log.setup`
```

Reply:
No, because the reasons pointed before. The last setup is the one that counts, they are not additive. Otherwise we are back to the `clear` method.

Reply:
Oh good. I thought this was like before where it would append backends!

Reply:
Oh wait so if you Log.setup and then later bind, will it append the backend like it does now, or overwrite the backend? I think it might be confusing if binding appended the backend instead of overwriting. 

But this is just a thought. Maybe I’m just overthinking it

Reply:
With the `Log.setup` methods we are actually able to hide `Log::Builder`. it will still work if you want to do `Log.builder.bind(...)` as long as there is no later `Log.setup` or `Log.builder.clear`